packages/protocol/test/util/fetch-mock.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Mock fetch response configuration
+ */
+export interface FetchMockResponse {
+    url: string | RegExp;
+    responseData: any;
+    options?: FetchMockResponseOptions;
+}
+
+/**
+ * Options for adding a mock response
+ */
+export interface FetchMockResponseOptions {
+    status?: number;
+    binary?: boolean;
+}
+
+/**
+ * Call log entry
+ */
+export interface FetchCallLogEntry {
+    url: string;
+    options: any;
+}
+
+/**
+ * Mock implementation of the global fetch API for testing purposes.
+ *
+ * This class allows you to intercept and mock fetch calls made during tests,
+ * making it possible to test code that makes HTTP requests without actually
+ * hitting real endpoints.
+ *
+ * @example
+ * ```typescript
+ * const fetchMock = new FetchMock();
+ * fetchMock.addResponse("/api/data", { result: "success" });
+ * fetchMock.install();
+ *
+ * // Your code that calls fetch("/api/data") will now receive the mocked response
+ *
+ * fetchMock.uninstall();
+ * ```
+ */
+export class FetchMock {
+    private responses: FetchMockResponse[] = [];
+    private originalFetch: typeof globalThis.fetch;
+    private callLog: FetchCallLogEntry[] = [];
+    private installed = false;
+
+    constructor() {
+        this.originalFetch = globalThis.fetch;
+    }
+
+    /**
+     * Add a mock response for a specific URL pattern.
+     *
+     * @param urlPattern - String or RegExp to match against request URLs
+     * @param responseData - Data to return (will be serialized to JSON or returned as binary)
+     * @param options - Optional configuration for status code and binary responses
+     */
+    addResponse(urlPattern: string | RegExp, responseData: any, options?: FetchMockResponseOptions) {
+        this.responses.push({
+            url: urlPattern,
+            responseData,
+            options,
+        });
+    }
+
+    /**
+     * Install this mock as the global fetch implementation.
+     * Must be called before the code under test executes fetch calls.
+     */
+    install() {
```

Comment:
We could also consider passing in `fetch` for components that require it.  Would also allow us to instrument